mantidproject/mantid
# Engineering Diffraction GUI - remove the need to select inst params file in GSAS tab

Previously, the GSAS tab of the ED GUI required the user to select an instrument parameter file (the file with DIFA, DIFC and TZERO constants). However it turns out this unnecessary, as we already calculated these in the Calibration tab and stored them in the main presenter. This PR removes the need for the user to go and find the file, and instead uses the file which we previously calculated.

**To test:**

Some sample data is [here](https://github.com/mantidproject/mantid/files/1894002/GSASIIRefineFitPeaks.zip) (I think there's also a `.prm` file in there but you can ignore it). You'll also need GSASII installed - go [here](http://docs.mantidproject.org/nightly/algorithms/GSASIIRefineFitPeaks-v1.html#installing-gsasii) for how to install GSASII.

If you've not opened the GUI before you'll be prompted to enter an RB number - just put anything in.

First, you'll need to run a calibration:

From the section **Make new calibration** in the Calibration tab, enter the run numbers for a vanadium and calibration (236516 and 241391 respectively are known to work). Click **Calibrate**. Go and make a cup of tea

Once the calibration has completed, go to the GSAS tab. You should see that the Instrument Parameter File input has disappeared. Choose one of the `.nxs` files for Focused Run File, all three `.cif` files for Phase Files, choose somewhere to create the `.gpx` file output by GSASII for New GSAS-II Project, and select the directory containing your GSASII installation (in particular the file `GSASIIscriptable.py`) for GSAS-II Installation Directory. Fiddle with any other parameters you like, then click **Run Refinement**. The fit should look reasonably sensible, and the algorithm should run with no issues.

Also a code review would be appreciated

No issue associated with this PR

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/commit/b036669c4b86fb5b922a2326a13d0e149aeb236b)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
